plugin/proxy/grpc_test.go

```diff
@@ -59,6 +64,97 @@ func TestStartupShutdown(t *testing.T) {
 	}
 }
 
+func TestRunAQuery(t *testing.T) {
+	grpclog.SetLogger(discard{})
+
+	upstream := &staticUpstream{
+		from: ".",
+		HealthCheck: healthcheck.HealthCheck{
+			Hosts: pool(),
+		},
+	}
+	g := newGrpcClient(nil, upstream)
+	upstream.ex = g
+
+	p := &Proxy{}
+	p.Upstreams = &[]Upstream{upstream}
+
+	err := g.OnStartup(p)
+	if err != nil {
+		t.Errorf("Error starting grpc client exchanger: %s", err)
+		return
+	}
+	// verify the client is usable, or an error is properly raised
+	state := request.Request{W: &test.ResponseWriter{}, Req: new(dns.Msg)}
+	g.Exchange(context.TODO(), "localhost:10053", state)
```

Comment:
`localhost:10053` is meant to *not* answer here? Fixed port numbers are a bad idea when testing

Reply:
Hum.. we need to know what is behind the address. Maybe it is better to create a Server, even though it is useless.